query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+            return jsonHandler;
+        }
+
+        /**
+         * Sets the {@link KeenJsonHandler} to use for handling JSON operations.
+         *
+         * @param jsonHandler The {@link KeenJsonHandler} to use.
+         */
+        public void setJsonHandler(KeenJsonHandler jsonHandler) {
+            this.jsonHandler = jsonHandler;
+        }
+
+        /**
+         * Sets the {@link KeenJsonHandler} to use for handling JSON operations.
+         *
+         * @param jsonHandler The {@link KeenJsonHandler} to use.
+         * @return This instance (for method chaining).
+         */
+        public QueryBuilder withJsonHandler(KeenJsonHandler jsonHandler) {
+            setJsonHandler(jsonHandler);
+            return this;
+        }
+
+        /**
+         * Gets the base URL to use for queries.
+         *
+         * @return The base URL to use.
+         */
+        public String getBaseURL() {return this.baseUrl;}
+
+        /**
+         * Sets the Base URL to use for queries.
+         *
+         * @param baseUrl The base URL to use.
+         */
+        public void setBaseUrl(String baseUrl) {this.baseUrl = baseUrl;}
+
+        /**
+         * Sets the Base URL to use for queries.
+         *
+         * @param baseUrl The base URL to use.
+         * @return This instance (for method chaining).
+         */
+        public QueryBuilder withBaseUrl(String baseUrl) {setBaseUrl(baseUrl); return this;}
+
+        /**
+         * Gets the {@link KeenProject} to use for queries.
+         *
+         * @return The {@link KeenProject}.
+         */
+        public KeenProject getKeenProject() {return this.project;}
+
+        /**
+         * Sets the {@link KeenProject} to use for queries.
+         *
+         * @param project The Keen Project containing Project ID and read/write keys.
+         */
+        public void setKeenProject(KeenProject project) {this.project = project;}
+
+        /**
+         * Builds a new Keen query client using the interfaces which have been specified explicitly on
+         * this builder instance via the set* or with* methods, or the default interfaces if none
+         * have been specified.
+         *
+         * @return A newly constructed Keen client.
+         */
+        public KeenQueryClient build() {
+
+            try {
+                if (httpHandler == null) {
+                    httpHandler = getDefaultHttpHandler();
+                }
+            } catch (Exception e) {
+                KeenLogging.log("Exception building HTTP handler: " + e.getMessage());
+            }
+
+            try {
+                if (jsonHandler == null) {
+                    jsonHandler = getDefaultJsonHandler();
+                }
+            } catch (Exception e) {
+                KeenLogging.log("Exception building JSON handler: " + e.getMessage());
+            }
+
+            if (project == null) {
+                project = new KeenProject("project","<readKey>", "<writeKey>");
```

Comment:
This was a bit of a garbled copy from the KeenClient.Builder code. Agreed. I think the right course of action for project == null is to throw an IllegalStateException, as the KeenClient does when there is no project specified and also no default project.